.devcontainer/devcontainer.json

```diff
@@ -0,0 +1,16 @@
+{
```

Comment:
The following should replicate the same environment but without the extra Dockerfile, conda, and the need to duplicate the requirements.txt file:
```
{
    "name": "Python 3",
	"image": "mcr.microsoft.com/devcontainers/python:3.10",
    "customizations": {
        "vscode":{
            "extensions": [
                "ms-python.python",
                "ms-python.black-formatter",
                "charliermarsh.ruff",
                "stkb.rewrap"
            ]
        }
    },
    "postCreateCommand": "pip install tox && pip install -e .[all]"
}
```
It still allows you to run `tox -e py310-test` to verify code changes work.

Reply:
Updated dev container! 👍

Reply:
@dr-rodriguez - unfortunately that new specification is not working nicely for me like the previous conda-based one was. When I come up in the terminal the path is not right:
```
@taldcroft ➜ /workspaces/astropy (samruddhikhandale/add-dev-container) $ which python
/usr/local/bin/python
@taldcroft ➜ /workspaces/astropy (samruddhikhandale/add-dev-container) $ which pip
/usr/local/bin/pip
```
I preferred the conda environment because that makes it easy for me to create a new environment with a different Python version. It happens sometimes that I need to debug something that only occurs with a particular Python version and conda is my go-to for that.

What I normally do is use conda to create the base environment, then make and activate an `astropy` conda env, then use `pip install -e .[all]` to get all the dependencies and astropy. That gives the best of both worlds IMO.

Reply:
Also with this change something is broken again with `setuptools_scm`:
```
@taldcroft ➜ /workspaces/astropy (samruddhikhandale/add-dev-container) $ which ipython
/home/vscode/.local/bin/ipython
@taldcroft ➜ /workspaces/astropy (samruddhikhandale/add-dev-container) $ ipython
Python 3.10.10 (main, Feb 11 2023, 02:40:33) [GCC 10.2.1 20210110]
Type 'copyright', 'credits' or 'license' for more information
IPython 8.11.0 -- An enhanced Interactive Python. Type '?' for help.

In [1]: import astropy

In [2]: astropy.__version__
Out[2]: '0.1.dev1+gb06e6cd'
```